apache/superset
# feat(native-filters): Hide non-numeric columns in numeric range filter

### SUMMARY
When user selects "Numerical range" native filter, only numeric columns should be available to choose. If there are no numeric columns in the dataset, show "No columns found" message in select column dropdown. When user has selected a non-numeric column and then changes filter type to numerical range, the column field should be reset.
We use `type_generic` field to check column's type. If a column doesn't have `type_generic` field, we assume that it's numeric as a failsafe.

Thank you @dpgaspar and @villebro for providing the backend code!

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
https://user-images.githubusercontent.com/15073128/123410870-15b3d700-d5b0-11eb-8895-99dd308fc558.mov

### TESTING INSTRUCTIONS
0. Enable `DASHBOARD_NATIVE_FILTERS` feature flag
1. Create a native filter and select "Numerical range" filter
2. Verify that only numeric columns are available to choose from (exception: some datasets from test data, like `message_channels`, have columns with `type_generic` field unset. In those cases, we display them as a failsafe)
3. Change filter type to Value and verify that all columns are available, not only numeric
4. Select some non-numeric column and change back to numerical range - column field should be reset.
5. 
### ADDITIONAL INFORMATION


- [x] Has associated issue: closes #15226
